File: src/main/ipc/claude.ts
import { ipcMain } from 'electron'
import * as fs from 'fs'
import * as os from 'os'
import * as path from 'path'

function hasClaudeSession(projectPath: string): boolean {
  const encoded = projectPath.replace(/\//g, '-')
  const sessionDir = path.join(os.homedir(), '.claude', 'projects', encoded)
  if (!fs.existsSync(sessionDir)) return false
  return fs.readdirSync(sessionDir).some(f => f.endsWith('.jsonl'))
}

export function registerClaudeHandlers(): void {
  ipcMain.handle('claude:hasSession', async (_event, projectPath: string) => {
    return hasClaudeSession(projectPath)
  })
}

File: src-tauri/src/commands/claude.rs
use std::path::Path;

fn encode_project_path_for_claude(project_path: &str) -> String {
    project_path.replace('/', "-")
}

#[tauri::command]
pub fn claude_has_session(project_path: String) -> bool {
    let home = dirs_fallback();
    let encoded = encode_project_path_for_claude(&project_path);
    let session_dir = Path::new(&home).join(".claude").join("projects").join(&encoded);

    if !session_dir.exists() { return false; }

    let Ok(entries) = std::fs::read_dir(&session_dir) else { return false };
    entries.flatten().any(|e| {
        e.file_name().to_string_lossy().ends_with(".jsonl")
    })
}

fn dirs_fallback() -> String {
    std::env::var("HOME").or_else(|_| std::env::var("USERPROFILE"))
        .unwrap_or_else(|_| "/home/unknown".to_string())
}
